wa-integrations/slack/routes/whatsapp.js

```diff
@@ -39,36 +48,91 @@ Router.route('/forwardWAMessage').post(async function (req, res) {
     }
   } else console.log('Contact found:', contact.name);
 
-  const url = 'https://slack.com/api/chat.postMessage';
-  const inputBody = {
-    channel: channel.channel_id,
-    text: `*${req.body.whatsapp.senderName}*: ${req.body.content.text}`,
-  };
-
-  if (!_.startsWith(contact.thread_ts, 'no thread assigned')) {
-    inputBody.thread_ts = contact.thread_ts;
-    console.log('Thread found');
+  if (req.body.content.contentType == "text") {
+    postTextMessage(channel, contact, req, res);
   }
 
-  try {
-    // Send request to Slack
-    const slackRes = await Axios.post(url, inputBody, { headers: slackHeaders });
-    console.log('Slack res:', slackRes.data);
-    if (_.startsWith(contact.thread_ts, 'no thread assigned')) {
-      Contacts.findOneAndUpdate({ phone: contact.phone }, { $set: { thread_ts: slackRes.data.ts } }, function (err, doc) {
-        if (err) {
-          res.status(500).send('DB error');
-          console.log('DB update error', err);
-        } else {
-          console.log('Thread parent saved:', slackRes.data.ts);
-        }
-      });
-    }
-    res.sendStatus(204);
-  } catch (error) {
-    res.sendStatus(500);
-    handleError(error);
+  if (req.body.content.contentType == 'media') {
+    postMediaMessage(channel, contact, req, res);
   }
 });
 
+async function postTextMessage (channel, contact, req, res) {
+  
+    const inputBody = {
+      channel: channel.channel_id,
+      text: `*${req.body.whatsapp.senderName}*: ${req.body.content.text}`,
+    };
+
+    if (!_.startsWith(contact.thread_ts, "no thread assigned")) {
+      inputBody.thread_ts = contact.thread_ts;
+      console.log('Thread found');
+    }
+
+    try {
+      // Send request to Slack
+      const slackRes = await Axios.post(postMessageUrl, inputBody, { headers: slackHeaders });
+      console.log('Slack res:', slackRes.data);
+      if (_.startsWith(contact.thread_ts, "no thread assigned")) {
+        Contacts.findOneAndUpdate({ phone: contact.phone }, { $set: { thread_ts: slackRes.data.ts } }, function (err, doc) {
+          if (err) {
+            res.status(500).send('DB error');
+            console.log('DB update error', err);
+          } else {
+            console.log('Thread parent saved:', slackRes.data.ts);
+          }
+        });
+      }
+      res.sendStatus(204);
+    } catch (error) {
+      res.sendStatus(500);
+      handleError(error);
+    }
+}
+
+async function postMediaMessage (channel, contact, req, res) {
```

Comment:
This is technically correct, however, we don't use the chaining functions to handle promises and rejections (.then, .catch), we use try/catch blocks to handle those. Please, consider rewriting it.